models/iaf_cond_alpha_mc_fixedca.cpp

```diff
@@ -0,0 +1,853 @@
+{
+  // we assume here that State_::get() always is called after
+  // Parameters_::get(),
+  // so that the per-compartment dictionaries exist
+  for ( size_t n = 0; n < NCOMP; ++n )
+  {
+    assert( d->known( comp_names_[ n ] ) );
+    DictionaryDatum dd = getValue< DictionaryDatum >( d, comp_names_[ n ] );
+
+    def< double >( dd, names::V_m, y_[ idx( n, V_M ) ] ); // Membrane potential
+  }
+}
+
+void
+nest::iaf_cond_alpha_mc_fixedca::State_::set( const DictionaryDatum& d,
+  const Parameters_& )
+{
+  // extract from sub-dictionaries
+  for ( size_t n = 0; n < NCOMP; ++n )
+    if ( d->known( comp_names_[ n ] ) )
+    {
+      DictionaryDatum dd = getValue< DictionaryDatum >( d, comp_names_[ n ] );
+      updateValue< double >( dd, names::V_m, y_[ idx( n, V_M ) ] );
+    }
+}
+
+
+/* ----------------------------------------------------------------
+ * Default and copy constructor for node, and destructor
+ * ---------------------------------------------------------------- */
+
+nest::iaf_cond_alpha_mc_fixedca::iaf_cond_alpha_mc_fixedca()
+  : Archiving_Node()
+  , P_()
+  , S_( P_ )
+  , B_( *this )
+{
+  recordablesMap_.create();
+
+  // set up table of compartment names
+  // comp_names_.resize(NCOMP); --- Fixed size, see comment on definition
+  comp_names_[ SOMA ] = Name( "soma" );
+  comp_names_[ PROX ] = Name( "proximal" );
+  comp_names_[ DIST ] = Name( "distal" );
+}
+
+nest::iaf_cond_alpha_mc_fixedca::iaf_cond_alpha_mc_fixedca(
+  const iaf_cond_alpha_mc_fixedca& n )
+  : Archiving_Node( n )
+  , P_( n.P_ )
+  , S_( n.S_ )
+  , B_( n.B_, *this )
+{
+}
+
+nest::iaf_cond_alpha_mc_fixedca::~iaf_cond_alpha_mc_fixedca()
+{
+  // GSL structs may not have been allocated, so we need to protect destruction
+  if ( B_.s_ )
+    gsl_odeiv_step_free( B_.s_ );
+  if ( B_.c_ )
+    gsl_odeiv_control_free( B_.c_ );
+  if ( B_.e_ )
+    gsl_odeiv_evolve_free( B_.e_ );
+}
+
+/* ----------------------------------------------------------------
+ * Node initialization functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::iaf_cond_alpha_mc_fixedca::init_state_( const Node& proto )
+{
+  const iaf_cond_alpha_mc_fixedca& pr =
+    downcast< iaf_cond_alpha_mc_fixedca >( proto );
+  S_ = pr.S_;
+}
+
+void
+nest::iaf_cond_alpha_mc_fixedca::init_buffers_()
+{
+  B_.spikes_.resize( NUM_SPIKE_RECEPTORS );
+  for ( size_t n = 0; n < NUM_SPIKE_RECEPTORS; ++n )
+    B_.spikes_[ n ].clear(); // includes resize
+
+  B_.currents_.resize( NUM_CURR_RECEPTORS );
+  for ( size_t n = 0; n < NUM_CURR_RECEPTORS; ++n )
+    B_.currents_[ n ].clear(); // includes resize
+
+  B_.logger_.reset();
+  Archiving_Node::clear_history();
+
+  B_.step_ = Time::get_resolution().get_ms();
+  B_.IntegrationStep_ = B_.step_;
+
+  if ( B_.s_ == 0 )
+    B_.s_ =
+      gsl_odeiv_step_alloc( gsl_odeiv_step_rkf45, State_::STATE_VEC_SIZE );
+  else
+    gsl_odeiv_step_reset( B_.s_ );
```

Comment:
You should use `{}` around all parts of an if-else, even if the contain only a single statement.